src/state/slices/playPausePlaylistSlice.ts

```diff
@@ -0,0 +1,35 @@
+import { createSlice, Dispatch } from '@reduxjs/toolkit'
+import { invoke } from '@tauri-apps/api'
+import { RootState } from 'state/store'
+
+export interface PlayPausePlaylistState {
+  value: boolean
+}
+
+const initialState = {
+  value: false
+} as PlayPausePlaylistState
+
+export const playPausePlaylistSlice = createSlice({
+  name: 'playlistPlaying',
+  initialState,
+  reducers: {
+    setPlaying: (state, action) => {
+      state.value = action.payload
+    }
+  }
+})
+
+export const { setPlaying } = playPausePlaylistSlice.actions
+
+export const togglePlay = () => (dispatch: Dispatch) => {
+  invoke('toggle_playlist', {}).then(() => {
+    invoke('get_playlist_playing', {}).then((data) => {
+      dispatch(setPlaying(data as boolean))
+    })
+  })
```

Comment:
can we change these to async/await?

Reply:
yeah, fire idea, i for some reason i thought async was slower but turns out its actually easier to implement optimizations with async/await... done